uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Form/ManageEntitlements.php

```diff
@@ -129,22 +129,33 @@ private function getContractsForCalculation(AbsencePeriod $absencePeriod) {
    */
   private function getActiveContractsForPeriod(AbsencePeriod $absencePeriod, $filter = []) {
     try {
-      $result = civicrm_api3('HRJobContract', 'getactivecontracts', [
-        'start_date' => $absencePeriod->start_date,
-        'end_date' => $absencePeriod->end_date,
-        'api.Contact.getvalue' => ['return' => 'display_name']
+      $contracts = CRM_Hrjobcontract_BAO_HRJobContract::getActiveContracts(
```

Comment:
regarding getActiveContracts() method ... are you aware that now only one active contract should be allowed at one time and you can get it by this method which I created before :
CRM_Hrjobcontract_BAO_HRJobContract::getCurrentContract($contactID)